x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/parser/LogicalPlanBuilder.java

```diff
@@ -523,9 +523,14 @@ public PlanFactory visitJoinCommand(EsqlBaseParser.JoinCommandContext ctx) {
         }
 
         var target = ctx.joinTarget();
+        var rightPattern = visitIndexPattern(List.of(target.index));
+        if (rightPattern.contains(WILDCARD)) {
```

Comment:
I guess here's where we can also put in the earliest possible validation against using remote indices as part of https://github.com/elastic/elasticsearch/pull/120277#issuecomment-2607164594.